pkg/deployment/deployment.go

```diff
@@ -101,14 +101,20 @@ func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
 
 	if name == "" {
 		builder.errorMsg = "deployment 'name' cannot be empty"
+
+		return &builder, fmt.Errorf(builder.errorMsg)
 	}
 
 	if nsname == "" {
 		builder.errorMsg = "deployment 'namespace' cannot be empty"
+
+		return &builder, fmt.Errorf(builder.errorMsg)
 	}
 
```

Comment:
Please add single verification and remove multiple return above.

	if builder.errorMsg != "" {
		return nil, fmt.Errorf("faield to pull deployment object due to the following error: %s", builder.errorMsg)
	}

Reply:
Ah yes I forgot to change this after we talked about the builder errorMsg structure.  I'll change that.

Reply:
I think the pattern should be if you set an `errorMsg`, you should immediately return.  Let's discuss!

Reply:
ok we can use multiple returns